chain/client/src/sync_jobs_actor.rs

```diff
@@ -1,74 +1,227 @@
-    pub(crate) actions: SyncJobsActions,
+// Set the mailbox capacity for the SyncJobsActor from default 16 to 100.
+const MAILBOX_CAPACITY: usize = 100;
+
+#[derive(Clone, MultiSend, MultiSenderFrom, MultiSendMessage)]
+#[multi_send_message_derive(Debug)]
+pub struct ClientSenderForSyncJobs {
+    apply_state_parts_response: Sender<ApplyStatePartsResponse>,
+    block_catch_up_response: Sender<BlockCatchUpResponse>,
+    resharding_response: Sender<ReshardingResponse>,
+    load_memtrie_response: Sender<LoadMemtrieResponse>,
 }
 
-impl SyncJobsActor {
-    pub(crate) const MAILBOX_CAPACITY: usize = 100;
+#[derive(Clone, MultiSend, MultiSenderFrom, MultiSendMessage)]
+#[multi_send_message_derive(Debug)]
+pub struct SyncJobsSenderForSyncJobs {
+    resharding_request: Sender<ReshardingRequest>,
 }
 
-impl actix::Actor for SyncJobsActor {
-    type Context = actix::Context<Self>;
+pub struct SyncJobsActor {
+    client_sender: ClientSenderForSyncJobs,
+    sync_jobs_sender: SyncJobsSenderForSyncJobs,
 }
 
-impl actix::Handler<WithSpanContext<LoadMemtrieRequest>> for SyncJobsActor {
-    type Result = ();
+impl Handler<LoadMemtrieRequest> for SyncJobsActor {
+    #[perf]
+    fn handle(&mut self, msg: LoadMemtrieRequest) {
+        self.handle_load_memtrie_request(msg);
+    }
+}
 
+impl Handler<ApplyStatePartsRequest> for SyncJobsActor {
     #[perf]
-    fn handle(
-        &mut self,
-        msg: WithSpanContext<LoadMemtrieRequest>,
-        _: &mut Self::Context,
-    ) -> Self::Result {
-        let (_span, msg) = handler_debug_span!(target: "client", msg);
-        self.actions.handle_load_memtrie_request(msg);
+    fn handle(&mut self, msg: ApplyStatePartsRequest) {
+        self.handle_apply_state_parts_request(msg);
     }
 }
 
-impl actix::Handler<WithSpanContext<ApplyStatePartsRequest>> for SyncJobsActor {
-    type Result = ();
+impl Handler<BlockCatchUpRequest> for SyncJobsActor {
+    #[perf]
+    fn handle(&mut self, msg: BlockCatchUpRequest) {
+        self.handle_block_catch_up_request(msg);
+    }
+}
 
+impl Handler<ReshardingRequest> for SyncJobsActor {
     #[perf]
-    fn handle(
-        &mut self,
-        msg: WithSpanContext<ApplyStatePartsRequest>,
-        _: &mut Self::Context,
-    ) -> Self::Result {
-        let (_span, msg) = handler_debug_span!(target: "client", msg);
-        self.actions.handle_apply_state_parts_request(msg);
+    fn handle(&mut self, msg: ReshardingRequest) {
+        self.handle_resharding_request(msg, &ActixFutureSpawner);
     }
 }
 
-impl actix::Handler<WithSpanContext<BlockCatchUpRequest>> for SyncJobsActor {
-    type Result = ();
+impl SyncJobsActor {
+    pub fn new(
+        client_sender: ClientSenderForSyncJobs,
+        sync_jobs_sender: SyncJobsSenderForSyncJobs,
+    ) -> Self {
+        Self { client_sender, sync_jobs_sender }
+    }
 
-    #[perf]
-    fn handle(
+    pub fn spawn_actix_actor(self) -> (actix::Addr<ActixWrapper<Self>>, actix::ArbiterHandle) {
```

Comment:
Now I'm not sure whether the best way to handle creating actix handler would be to have functions like this or to write something generic?

I'm guessing each handler would have it's own specific requirements? Like for example, here we need to set the mailbox capacity which is actix specific.

Reply:
It's currently adhoc, so we can do it adhoc for now. I'm not worried about it as long as it works.